File: Builder.py
class House:
    def __init__(self, rooms, bathrooms, garage, pool, garden):
        self.rooms = rooms
        self.bathrooms = bathrooms
        self.garage = garage
        self.pool = pool
        self.garden = garden

    def get_details(self):
        return {
            "Rooms": self.rooms,
            "Bathrooms": self.bathrooms,
            "Garage": self.garage,
            "Pool": self.pool,
            "Garden": self.garden
        }

class HouseBuilder:
    def __init__(self):
        self.rooms = 0
        self.bathrooms = 0
        self.garage = False
        self.pool = False
        self.garden = False

    def set_rooms(self, rooms):
        self.rooms = rooms
        return self

    def set_bathrooms(self, bathrooms):
        self.bathrooms = bathrooms
        return self

    def add_garage(self):
        self.garage = True
        return self

    def add_pool(self):
        self.pool = True
        return self

    def add_garden(self):
        self.garden = True
        return self

    def build(self):
        return House(self.rooms, self.bathrooms, self.garage, self.pool, self.garden)


house = HouseBuilder().set_rooms(3).set_bathrooms(1).add_garage().add_garden().build()

print(house.get_details())

File: builder.cpp
#include <iostream>
#include <string>
using namespace std;

class House {
public:
    int rooms;
    int bathrooms;
    bool garage;
    bool pool;
    bool garden;

    House(int rooms, int bathrooms, bool garage, bool pool, bool garden) {
        this->rooms = rooms;
        this->bathrooms = bathrooms;
        this->garage = garage;
        this->pool = pool;
        this->garden = garden;
    }

    void get_details() {
        cout << "Rooms: " << rooms <<endl;
        cout << "Bathrooms: " << bathrooms <<endl;
        cout << "Garage: " << (garage ? "Yes" : "No") <<endl;
        cout << "Pool: " << (pool ? "Yes" : "No") <<endl;
        cout << "Garden: " << (garden ? "Yes" : "No") <<endl;
    }
};

class HouseBuilder {
private:
    int rooms;
    int bathrooms;
    bool garage;
    bool pool;
    bool garden;

public:
    HouseBuilder() : rooms(0), bathrooms(0), garage(false), pool(false), garden(false) {}

    HouseBuilder& set_rooms(int rooms) {
        this->rooms = rooms;
        return *this;
    }

    HouseBuilder& set_bathrooms(int bathrooms) {
        this->bathrooms = bathrooms;
        return *this;
    }

    HouseBuilder& add_garage() {
        this->garage = true;
        return *this;
    }

    HouseBuilder& add_pool() {
        this->pool = true;
        return *this;
    }

    HouseBuilder& add_garden() {
        this->garden = true;
        return *this;
    }

    House build() {
        return House(rooms, bathrooms, garage, pool, garden);
    }
};

int main() {
    House house = HouseBuilder().set_rooms(3).set_bathrooms(1).add_garage().add_garden().build();

    house.get_details();

    return 0;
}
